app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/VaultItemListingScreen.kt

```diff
@@ -137,6 +144,30 @@ fun VaultItemListingScreen(
             is VaultItemListingEvent.CompleteFido2Registration -> {
                 fido2CompletionManager.completeFido2Registration(event.result)
             }
+
+            is VaultItemListingEvent.Fido2UserVerification -> {
+                if (biometricsManager.isUserVerificationSupported) {
+                    biometricsManager.promptUserVerification(
+                        onSuccess = {
+                            userVerificationHandlers
+                                .onUserVerificationSuccess(event.selectedCipherView)
+                        },
+                        onCancel = userVerificationHandlers.onUserVerificationCancelled,
+                        onLockOut = userVerificationHandlers.onUserVerificationLockOut,
+                        onError = userVerificationHandlers.onUserVerificationFail,
+                    )
+                } else {
+                    if (event.isRequired) {
```

Comment:
Can we just make this an `else if`

Reply:
I am also curious if some of this logic can be done elsewhere, it is a bit weird to have here when processing an event.

Reply:
We can move it into `biometricsManager.promptUserVerification` if we add `isRequired` as an argument.

Reply:
Took a different approach. Added a callback explicitly for `onNotSupported` and corresponding view model action.